packages/core/contracts/bot-helpers/atomic-disputer/ReserveCurrencyDisputer.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "@uniswap/lib/contracts/libraries/TransferHelper.sol";
+import "@uniswap/v2-periphery/contracts/interfaces/IUniswapV2Router01.sol";
+
+import "../../common/implementation/FixedPoint.sol";
+
+/**
+ * @title ReserveCurrencyDisputer
+ * @notice Helper contract to enable a disputer to hold one reserver currency and dispute against any number of
+ * financial contracts. Is assumed to be called by a DSProxy which holds reserve currency.
+ */
+
+contract ReserveCurrencyDisputer {
```

Comment:
tiny nit: does this work out of the box for both uniswap v2 and v3? If not, should we name this something that implies it's v2-specific as we've done retroactively for some of the trader functionality?

Reply:
it does not work with uniswap v3! this is basically the same as the single reserve currency liquidator in that it only works with v2.

it might be worth us considering how we want to consolidate having multiple implementations of things, such as this reserve liquidator/disputer/trader that operate over different market versions. The simplest solution might be to just _leave_ this as being uni v2 support only until uni v3 has grown enough. Once there is enough liquidity in all the pairs we care about on v3 we can simply upgrade and deprecate the old contracts so we don't need to support two sets. WDYT?